_posts/2015-11-12-Using-OCR-To-Fix-A-Hilarious-Bug.markdown

```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title: 'Using OCR To Fix a Hilarious Big'
+date: 2015-11-05T00:00:00.000Z
+comments: false
+categories: [Ruby, open source, OSS, debugging]
+author: matt
+---
+
+For a little while, we would get very strange bug reports. People would complain that artist thumbnails (viewed in several different contexts across the web and our iOS apps) would not be an image of the artist's work, but rather text, which had inexplicably become an actual JPG. This wasn't just text appearing in a `div` that should contain an `img` or something like that, these were actual JPG's that were pictures of text.
+
+We would fix these as they came up, chalking the strangeness up to some relic of an old image processing pipeline, data being migrated, etc.
+
+However, the reports kept coming in. This blog post is about how we diagnosed this actual bug (linked to 'features' of two different pieces of software we use to manage our data- the fix was then trivial), and how we used a simple Ruby script and OCR to help us detect and fix the existing images.
+
+<!-- more -->
+
+Here's an example of a bug report where the thumbnail for [Marina Abramović](https://www.artsy.net/artist/marina-abramovic-1) became the text of her bio.
+
+![Bad Search](/images/2015-11-12-hilarious-bug/search.png)
+
+Here's one from our [iOS app](https://github.com/artsy/eigen) showing that thumbnails for related artists are set to their bios as well.
+
+![Bad Related Artists](/images/2015-11-12-hilarious-bug/eigen.png)
+
+Weird right? We eventually tracked down what was going on, and it's actually perfectly summarized in [this issue](https://github.com/blueimp/jQuery-File-Upload/pull/3356).
+
+We have an admin panel that allows some metadata about an artist to be edited. This includes their bio, as well as a place to upload a representative image as their 'cover thumbnail'.
+
+As the issue describes, we had some text input fields, as well as a file upload form using [Blueimp's jQuery File Upload](https://github.com/blueimp/jQuery-File-Upload). When you don't specify a `pasteZone` it defaults to the entire document. This means that a paste event anywhere on the page will trigger that event.
+
+Our [Art Genome Project Team](https://www.artsy.net/theartgenomeproject), which manages artist metadata, was using Microsoft Excel and Word to organize some of that data, including bios. When ready, a team member would copy and paste the bio into the bio input text field. This would also immediately fire the event for the image upload, which now automagically became an actual picture of the text of the bio. Our API and image processing pipeline would happily accept that, leading to the incredibly bizarre bug reports.
+
+My immediate fix was to specify and scope `pasteZone` (and similarly, `dropZone`) to the element the file upload widget was bound to, and try to figure out a way to programmaticaly detect these problematic images.
```

Comment:
There's a leap here. It's _not_ obvious that one should take a programmatic approach to fix this strange bug. I'd make this a separate sentence and either convey how _many_ artists it impacted (e.g., thousands, justifying the programmatic fix), or how it was primarily curiosity about what was possible that drove you in this direction.